### PR TITLE
ci(release): Fetch the last 20 commits, not just 1

### PR DESCRIPTION
**Pull request by BYK**
This should fix the merge error we encountered in the last release: https://github.com/getsentry/sentry/runs/1413845407?check_suite_focus=true#step:10:41

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -46,6 +46,7 @@ jobs:
       - uses: actions/checkout@v2
         with:
           token: ${{ secrets.GH_SENTRY_BOT_PAT }}
+          fetch-depth: 20
```
Comment (chadwhitacre) [inline .github/workflows/release.yml:49]:
```
    # Number of commits to fetch. 0 indicates all history for all branches and tags.
    # Default: 1
    fetch-depth: ''
```

https://github.com/actions/checkout/tree/5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f

**Comment by chadwhitacre** [comment]
Remind me ... how does increasing the fetch depth resolve the linked error?

```
Already on 'master'
Your branch is up to date with 'origin/master'.
New version: 20.12.0.dev0
error: insufficient permission for adding an object to repository database .git/objects
error: LICENSE: failed to insert into database
error: unable to index file 'LICENSE'
fatal: updating files failed
Error: Process completed with exit code 128.
```

**Comment by BYK** [comment]
@chadwhitacre couldn't find the exact place I read this but when you have only 1 commit, `git` needs the other commit if you try to merge and I think `checkout@v2` runs with different permissions than our userspace commands so git fails to do this. Or this is just because of the missing history and `git` assumes things to be there.

**Comment by chadwhitacre** [comment]
Okay, and this is a case where we're essentially testing in prod, yes? Or were you able to test this outside of CI?

**Comment by BYK** [comment]
>Or were you able to test this outside of CI?

You know what, I _can_ actually test this out. I'll do that and come back here with results.

**Comment by BYK** [comment]
Just verified that `.git/objects` is empty on a fresh checkout with `--depth 1` and gets populated with more folders once I make a change in `LICENSE` and commit that. This still doesn't guarantee that the issue will be resolved but I think is enough evidence that this may work so let's go ahead and merge this.

**Review by chadwhitacre — APPROVED** [review]
I'd rather see this tested somewhere besides `master`, but it seems we're not set up for that.

**Comment by chadwhitacre** [comment]
My hunch is that changing the fetch depth won't fix the permissions issue, because In my testing I don't see that cloning with depth > 1 affects what hex subdirs show up in `.git/objects`:

```
$ git clone git@github.com:getsentry/onpremise.git --depth=20
Cloning into 'onpremise'...
remote: Enumerating objects: 120, done.
remote: Counting objects: 100% (120/120), done.
remote: Compressing objects: 100% (84/84), done.
Receiving objects: 100% (120/120), 44.14 KiB | 1.92 MiB/s, done.
remote: Total 120 (delta 50), reused 79 (delta 28), pack-reused 0
Resolving deltas: 100% (50/50), done.
$ ls -FGl onpremise/.git/objects 
total 0
drwxr-xr-x  2 chadwhitacre  staff   64 Nov 25 14:04 info/
drwxr-xr-x  4 chadwhitacre  staff  128 Nov 25 14:04 pack/
$ rm -rf onpremise 
$ git clone git@github.com:getsentry/onpremise.git --depth=1 
Cloning into 'onpremise'...
remote: Enumerating objects: 40, done.
remote: Counting objects: 100% (40/40), done.
remote: Compressing objects: 100% (33/33), done.
remote: Total 40 (delta 0), reused 24 (delta 0), pack-reused 0
Receiving objects: 100% (40/40), 27.45 KiB | 3.92 MiB/s, done.
$ ls -FGl onpremise/.git/objects                            
total 0
drwxr-xr-x  2 chadwhitacre  staff   64 Nov 25 14:06 info/
drwxr-xr-x  4 chadwhitacre  staff  128 Nov 25 14:06 pack/
$
```

**Comment by chadwhitacre** [comment]
I'd rather understand why there's a perms delta between the checkout step and the version bump step, and look for a way to resolve that directly. What users are in play in each?

That said, don't let me get in the way if you want to merge this in order to test the hypothesis that fetch depth will help.

**Comment by BYK** [comment]
>I'd rather understand why there's a perms delta between the checkout step and the version bump step, and look for a way to resolve that directly. What users are in play in each?

This is a GitHub Actions mystery that I'm not too interested in uncovering right now 😀

**Pull request merged**